File: frontend/src-tauri/src/analytics/analytics.rs
// Analytics module - no-op stubs (PostHog telemetry removed for privacy)

use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnalyticsConfig {
    pub api_key: String,
    pub host: Option<String>,
    pub enabled: bool,
}

impl Default for AnalyticsConfig {
    fn default() -> Self {
        Self {
            api_key: String::new(),
            host: None,
            enabled: false,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UserSession {
    pub session_id: String,
    pub user_id: String,
    pub start_time: chrono::DateTime<chrono::Utc>,
    pub is_active: bool,
}

impl UserSession {
    pub fn new(user_id: String) -> Self {
        Self {
            session_id: format!("session_{}", uuid::Uuid::new_v4()),
            user_id,
            start_time: chrono::Utc::now(),
            is_active: false,
        }
    }

    pub fn duration_seconds(&self) -> i64 {
        (chrono::Utc::now() - self.start_time).num_seconds()
    }
}

pub struct AnalyticsClient;

impl AnalyticsClient {
    pub async fn new(_config: AnalyticsConfig) -> Self {
        Self
    }

    pub async fn identify(&self, _user_id: String, _properties: Option<HashMap<String, String>>) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_event(&self, _event_name: &str, _properties: Option<HashMap<String, String>>) -> Result<(), String> {
        Ok(())
    }

    pub async fn start_session(&self, _user_id: String) -> Result<String, String> {
        Ok(String::new())
    }

    pub async fn end_session(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_daily_active_user(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_user_first_launch(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn get_current_session(&self) -> Option<UserSession> {
        None
    }

    pub async fn is_session_active(&self) -> bool {
        false
    }

    pub async fn track_meeting_started(&self, _meeting_id: &str, _meeting_title: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_recording_started(&self, _meeting_id: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_recording_stopped(&self, _meeting_id: &str, _duration_seconds: Option<u64>) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_meeting_deleted(&self, _meeting_id: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_settings_changed(&self, _setting_type: &str, _new_value: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_app_started(&self, _version: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_feature_used(&self, _feature_name: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_generation_started(&self, _model_provider: &str, _model_name: &str, _transcript_length: usize) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_generation_completed(&self, _model_provider: &str, _model_name: &str, _success: bool, _duration_seconds: Option<u64>, _error_message: Option<&str>) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_regenerated(&self, _model_provider: &str, _model_name: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_model_changed(&self, _old_provider: &str, _old_model: &str, _new_provider: &str, _new_model: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_custom_prompt_used(&self, _prompt_length: usize) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_meeting_ended(
        &self,
        _transcription_provider: &str,
        _transcription_model: &str,
        _summary_provider: &str,
        _summary_model: &str,
        _total_duration_seconds: Option<f64>,
        _active_duration_seconds: f64,
        _pause_duration_seconds: f64,
        _microphone_device_type: &str,
        _system_audio_device_type: &str,
        _chunks_processed: u64,
        _transcript_segments_count: u64,
        _had_fatal_error: bool,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_enabled(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_disabled(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_transparency_viewed(&self) -> Result<(), String> {
        Ok(())
    }

    pub fn is_enabled(&self) -> bool {
        false
    }

    pub async fn set_user_properties(&self, _properties: HashMap<String, String>) -> Result<(), String> {
        Ok(())
    }
}

// Helper function to create analytics client from config
pub async fn create_analytics_client(config: AnalyticsConfig) -> AnalyticsClient {
    AnalyticsClient::new(config).await
}

File: frontend/src/lib/analytics.ts
export interface AnalyticsProperties {
  [key: string]: string;
}

export interface DeviceInfo {
  platform: string;
  os_version: string;
  architecture: string;
}

export interface UserSession {
  session_id: string;
  user_id: string;
  start_time: string;
  last_heartbeat: string;
  is_active: boolean;
}

const DEFAULT_DEVICE_INFO: DeviceInfo = {
  platform: 'unknown',
  os_version: 'unknown',
  architecture: 'unknown',
};

export class Analytics {
  static async init(): Promise<void> {}
  static async disable(): Promise<void> {}
  static async isEnabled(): Promise<boolean> { return false; }
  static async track(_eventName: string, _properties?: AnalyticsProperties): Promise<void> {}
  static async identify(_userId: string, _properties?: AnalyticsProperties): Promise<void> {}
  static async startSession(_userId: string): Promise<string | null> { return null; }
  static async endSession(): Promise<void> {}
  static async trackDailyActiveUser(): Promise<void> {}
  static async trackUserFirstLaunch(): Promise<void> {}
  static async isSessionActive(): Promise<boolean> { return false; }
  static async getPersistentUserId(): Promise<string> { return 'telemetry-disabled'; }
  static async checkAndTrackFirstLaunch(): Promise<void> {}
  static async checkAndTrackDailyUsage(): Promise<void> {}
  static getCurrentUserId(): string | null { return null; }
  static async getPlatform(): Promise<string> { return DEFAULT_DEVICE_INFO.platform; }
  static async getOSVersion(): Promise<string> { return DEFAULT_DEVICE_INFO.os_version; }
  static async getArchitecture(): Promise<string> { return DEFAULT_DEVICE_INFO.architecture; }
  static async getDeviceInfo(): Promise<DeviceInfo> { return DEFAULT_DEVICE_INFO; }
  static async trackSessionStarted(_sessionId: string): Promise<void> {}
  static async trackSessionEnded(_sessionId: string): Promise<void> {}
  static async trackMeetingCompleted(
    _meetingId: string,
    _metrics: {
      duration_seconds: number;
      transcript_segments: number;
      transcript_word_count: number;
      words_per_minute: number;
      meetings_today: number;
    },
  ): Promise<void> {}
  static async trackFeatureUsedEnhanced(_featureName: string, _properties?: Record<string, unknown>): Promise<void> {}
  static async trackCopy(_copyType: 'transcript' | 'summary', _properties?: Record<string, unknown>): Promise<void> {}
  static async trackMeetingStarted(_meetingId: string, _meetingTitle: string): Promise<void> {}
  static async trackRecordingStarted(_meetingId: string): Promise<void> {}
  static async trackRecordingStopped(_meetingId: string, _durationSeconds?: number): Promise<void> {}
  static async trackMeetingDeleted(_meetingId: string): Promise<void> {}
  static async trackSettingsChanged(_settingType: string, _newValue: string): Promise<void> {}
  static async trackFeatureUsed(_featureName: string): Promise<void> {}
  static async trackPageView(_pageName: string): Promise<void> {}
  static async trackButtonClick(_buttonName: string, _location?: string): Promise<void> {}
  static async trackError(_errorType: string, _errorMessage: string): Promise<void> {}
  static async trackAppStarted(): Promise<void> {}
  static async cleanup(): Promise<void> {}
  static reset(): void {}
  static async waitForInitialization(_timeout = 5000): Promise<boolean> { return false; }
  static async trackBackendConnection(_success: boolean, _error?: string): Promise<void> {}
  static async trackTranscriptionError(_errorMessage: string): Promise<void> {}
  static async trackTranscriptionSuccess(_duration?: number): Promise<void> {}
  static async trackSummaryGenerationStarted(
    _modelProvider: string,
    _modelName: string,
    _transcriptLength: number,
    _timeSinceRecordingMinutes?: number,
  ): Promise<void> {}
  static async trackSummaryGenerationCompleted(
    _modelProvider: string,
    _modelName: string,
    _success: boolean,
    _durationSeconds?: number,
    _errorMessage?: string,
  ): Promise<void> {}
  static async trackSummaryRegenerated(_modelProvider: string, _modelName: string): Promise<void> {}
  static async trackModelChanged(
    _oldProvider: string,
    _oldModel: string,
    _newProvider: string,
    _newModel: string,
  ): Promise<void> {}
  static async trackCustomPromptUsed(_promptLength: number): Promise<void> {}
  static async getMeetingsCountToday(): Promise<number> { return 0; }
  static async updateMeetingCount(): Promise<void> {}
  static async calculateDaysSince(_key: string): Promise<number> { return 0; }
}

export default Analytics;
